### PR TITLE
Added Gold Price Detector

### LINKED ISSUE #72: [New Model] Gold Price Prediction
### Problem Description:
The gold price is influenced by a variety of economic factors, including stock market performance, crude oil prices, silver prices, and currency exchange rates. Predicting gold prices has become increasingly important for investors and financial analysts looking to make informed decisions in the commodities market. This project aims to build a machine learning model that can predict gold prices using key financial indicators such as the S&P 500 index (SPX), crude oil prices (USO), silver prices (SLV), and the EUR/USD exchange rate.

By analyzing these factors, we can gain valuable insights into how fluctuations in these markets influence gold prices, allowing for more accurate forecasting.

### Model Description:
The gold price prediction model will be built using Linear Regression. The model will take historical data from financial indicators (SPX, USO, SLV, EUR/USD) as input features to predict future gold prices (GLD). The chosen model is well-suited for capturing the relationships between these financial variables and gold prices, enabling accurate forecasting.

### Estimated Time for Completion:
The project is expected to take approximately 2-3 days to complete. This includes the following steps:

- Data Collection & Preprocessing: Loading the gold price dataset, handling missing values, and normalizing the data.
- Model Training & Validation: Training the Random Forest model, validating its performance, and fine-tuning the model for optimal accuracy.
- Model Evaluation: Assessing the model’s performance using metrics like Mean Squared Error (MSE) and R-squared score.

### Expected Outcome:
The outcome of this project will be a machine learning model capable of accurately predicting gold prices based on key economic indicators. The model will help financial analysts and investors forecast gold price trends and make informed decisions, improving their ability to hedge risks and seize market opportunities. Additionally, the model's insights into how different financial factors interact with gold prices will enhance our understanding of the gold market's behavior.

### PR DESCRIPTION
**Pull request by J-B-Mugundh**
### Description
This pull request adds a feature to predict gold prices (GLD) based on other financial metrics such as SPX, USO, SLV, and EUR/USD. The prediction is powered by Random Forest model loaded using joblib.

### Issue Resolved
#72 

### Changes Made
Added necessary files as mentioned in the guidelines

### Screenshots or Videos
Image

### Checklist

- [x] My code follows the current [project structure](https://github.com/yashasvini121/predictive-calc/blob/master/docs/project-structure.md)
- [x] I have thoroughly reviewed and updated the `requirements.txt` file to include any new packages
- [x] I have added relevant tests (if necessary).
- [x] I have added comments in the code where needed.
- [x] This PR is submitted under **Hacktoberfest**.
- [x] This PR is submitted under **GirlScript Summer of Code (GSSoC-Extd)**.

### CODE REVIEW

**Comment by J-B-Mugundh** [comment]
@yashasvini121 Please review the PR and let me know your comments!

File: form_configs/gold_price_prediction.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "Gold Price Form": {
+    "SPX": {
+      "type": "number",
+      "min_value": 0.0,
+      "default_value": 1447.160034,
```
Comment (yashasvini121) [inline form_configs/gold_price_prediction.json:6]: The values on the forms are rounded off to the last 2 digits, so the updates in the fields aren't noticeable

File: pages/pages.json [hunk 2]
```diff
@@ -93,6 +93,26 @@
...
+    "tabs": [
+      {
+        "name": "Gold Price Form",
+        "type": "form",
+        "form_name": "Gold Price Form"
+      },
+      {
+        "name": "Model Details",
+        "type": "model_details"
+      }
```
Comment (yashasvini121) [inline pages/pages.json:114]: No need to write Model details tab if it's empty.

File: models/gold_price_prediction/notebooks/Gold_Price_Prediction.ipynb
Comment (yashasvini121) [inline models/gold_price_prediction/notebooks/Gold_Price_Prediction.ipynb:1]: notebook needs to have a `model_details()` function that prints the model details report.

**Review by yashasvini121 — CHANGES_REQUESTED** [review]
Add your model description to `App.py`. Also explain all the terms that are in the form in `st.expander()`

**Comment by J-B-Mugundh** [comment]
Updated! Please check

**Review by yashasvini121 — APPROVED** [review]

**Pull request merged**